### PR TITLE
gcp bucket update 

### PR DESCRIPTION
**Pull request by fjvela**
# Description

Update GCP Storage Bucket binding:

- update create operation: support upload files in base64 (https://docs.dapr.io/reference/components-reference/supported-bindings/gcpbucket/#upload-a-file doesn't work), return object url to download
- add get operation
- add delete operation
- add list operation
- create operation replace field name with field key (field name backward compatibility), now you can use blob storage, s3 or gcp bucket without change metadata request 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/components-contrib/issues/1125

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: https://github.com/dapr/docs/issues/1790

### CODE REVIEW

**Comment by yaron2** [comment]
@beiwei30 can you please review this?

**Comment by yaron2** [comment]
@fjvela please resolve the conflict and we'll move on to merging. thanks!

File: bindings/gcp/bucket/bucket.go [hunk 1]
```diff
@@ -45,62 +74,237 @@ func NewGCPStorage(logger logger.Logger) *GCPStorage {
...
+
+	return &bindings.InvokeResponse{
+		Data: b,
+	}, nil
+}
+
+func (g *GCPStorage) get(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata, err := g.metadata.mergeWithRequestMetadata(req)
+	if err != nil {
+		return nil, fmt.Errorf("s3 binding error. error merge metadata : %w", err)
```
Comment (beiwei30) [inline bindings/gcp/bucket/bucket.go:192]: Should this error msg be 'gcp binding error' instead of 's3‘?

**Review by beiwei30 — CHANGES_REQUESTED** [review]
@fjvela Javier, your change looks good to me except for this minor issue.

**Review by beiwei30 — APPROVED** [review]
LGTM now, Thanks Javier for this effort.

**Comment by yaron2** [comment]
Merged. Thank you @fjvela for your contribution, and thank you @beiwei30 for the review.

**Pull request merged**